modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js

```diff
@@ -12,449 +12,337 @@
-					document.activeElement.blur();
-				}
-			}
-			else {
-				newValue = this.addValue(itemValue);
-			}
-		}
-		else {
-			newValue = this.setValue(itemValue);
-		}
-
-		preventDefault();
-
-		this.setState(
-			{
-				expanded: multiple,
-				value: newValue,
-			},
-			() =>
-				this.emit('fieldEdited', {
-					fieldInstance: this,
-					value: newValue,
-				})
+const Trigger = forwardRef(
+	({onCloseButtonClicked, onTriggerClicked, value, ...otherProps}, ref) => {
+		return (
+			<>
+				<HiddenSelectInput value={value} {...otherProps} />
+				<VisibleSelectInput
+					onClick={onTriggerClicked}
+					onCloseButtonClicked={onCloseButtonClicked}
+					ref={ref}
+					value={value}
+					{...otherProps}
+				/>
+			</>
 		);
 	}
+);
+
+const Select = ({
+	multiple,
+	onCloseButtonClicked,
+	onDropdownItemClicked,
+	onExpand,
+	options,
+	predefinedValue,
+	readOnly,
+	value,
+	...otherProps
+}) => {
+	const menuElementRef = useRef(null);
+	const triggerElementRef = useRef(null);
+
+	const [currentValue, setCurrentValue] = useState(value);
+	const [expand, setExpand] = useState(false);
+
+	return (
+		<div>
+			<Trigger
+				onCloseButtonClicked={({event, value}) => {
+					const newValue = removeValue({
+						value: currentValue,
+						valueToBeRemoved: value,
+					});
+
+					setCurrentValue(newValue);
+
+					onCloseButtonClicked({event, value: newValue});
+				}}
+				onTriggerClicked={event => {
+					event.preventDefault();
+
+					if (readOnly) {
+						return;
+					}
+
+					triggerElementRef.current = event.target;
+
+					setExpand(!expand);
+					onExpand({event, expand: !expand});
+				}}
+				options={options}
+				predefinedValue={predefinedValue}
+				readOnly={readOnly}
+				value={currentValue}
+				{...otherProps}
+			/>
+			<ClayDropDown.Menu
+				active={expand}
+				alignElementRef={triggerElementRef}
+				className="ddm-btn-full ddm-select-dropdown"
+				onSetActive={setExpand}
+				ref={menuElementRef}
+			>
+				<ClayDropDown.ItemList>
+					{options.map((option, index) => (
+						<ClayDropDown.Item
+							active={expand && currentValue === option.label}
+							className="ddm-btn-full ddm-select-dropdown"
+							data-testid={`dropdownItem-${index}`}
+							key={`dropdown-option-${index}`}
+							label={option.label}
+							onClick={event => {
+								const itemValue = option.value;
+
+								let newValue;
+
+								if (multiple) {
+									if (currentValue.includes(itemValue)) {
+										newValue = removeValue({
+											value: currentValue,
+											valueToBeRemoved: itemValue,
+										});
+
+										// Forces the active element to be blurred.
+										if (document.activeElement) {
+											document.activeElement.blur();
+										}
+									}
+									else {
+										newValue = appendValue({
+											value: currentValue,
+											valueToBeAppended: itemValue,
+										});
+									}
+								}
+								else {
+									newValue = toArray(itemValue);
```

Comment:
instead of `toArray` just one `[itemValue] `is enough.

Reply:
✅